modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -402,6 +417,28 @@ protected Experiment internalGetExperiment(Application.Name appName,
     }
 
     /**
+     * Updates the tags for a given Application.
+     *
+     * @param applicationName the Application for which the tags should be added
+     * @param tags            list of tags for the experiment
+     */
+    protected void updateExperimentTags(Application.Name applicationName, Set<String> tags) {
+
+        if (tags == null)
+            tags = Collections.EMPTY_SET;
+
+        // update the tag table, just rewrite the complete entry
+        List<ExperimentTagsByApplication> listAllTags = experimentAccessor.getAllTags(applicationName.toString()).all();
```

Comment:
Hm so first I think the delete makes sense, because you use it in the filtering completion and that only makes sense for existing tags. It also allows Users to make mistakes and not clutter their applications history forever.

And I understand your concerns concerning the dirty read/write and I think your suggestion is good. I just don't feel it to be so crucial for this feature. In the unlikely event of a simultaneous change to the tags of different experiments, the worst thing to happen is a wrong suggestion in the auto-complete on the search. But it can be changed of course - it will just take a bit of time :D